src/compile.ts

```diff
@@ -407,6 +472,291 @@ export function compile(
         return call;
       }
 
+      function visitFunction(
+        func: FunctionInterface,
+        context: ts.TransformationContext
+      ): ts.Node {
+        const [_one, _two, funcDecl, _four] = func.arguments;
+
+        return ts.factory.updateNewExpression(
+          func,
+          func.expression,
+          func.typeArguments,
+          [
+            _one,
+            _two,
+            errorBoundary(() => toNativeFunction(funcDecl, context)),
+            ...(_four ? [_four] : []),
+          ]
+        );
+      }
+
+      interface NativeExprContext {
+        preWarmContext: ts.Identifier;
+        closureNode: TsFunctionParameter;
+        registerIntegration: (node: ts.Expression) => void;
+      }
+
+      /**
+       * Native Functions do not allow the creation of resources, CDK constructs or functionless.
+       */
+      function toNativeFunction(
```

Comment:
Provide a comment that shows an example of how the code is transformed. I'm particularly interested in preWarmContext.

Reply:
Done